packages/frontend/src/hooks/echarts/useEcharts.ts

```diff
@@ -1269,6 +1270,61 @@ const useEcharts = () => {
               )
             : allColors;
     }, [organizationData?.chartColors, validCartesianConfig]);
+    const sortedResults = useMemo(() => {
+        const results =
+            validCartesianConfig?.layout?.xField === EMPTY_X_AXIS
+                ? getResultValueArray(rows, true).map((s) => ({
+                      ...s,
+                      [EMPTY_X_AXIS]: ' ',
+                  }))
+                : getResultValueArray(rows, true);
+        try {
+            if (!explore) return results;
+            const dimensions = getDimensions(explore);
+
+            const xFieldId = validCartesianConfig?.layout?.xField;
+            if (xFieldId === undefined) return results;
+
+            const alreadySorted = resultsData?.metricQuery.sorts?.find(
+                (sort) => sort.fieldId === xFieldId,
+            );
```

Comment:
We need to check if the first sort matches the x field id.
```suggestion
            const alreadySorted = resultsData?.metricQuery.sorts?.[0].fieldId ==- xFieldId;
```


Example of data that is sorted first by name and date second:
```
jose, 2020-10-03
jose, 2020-10-04
javi, 2020-10-01 
javi, 2020-10-02 
```